test/random_dem.cpp

```diff
@@ -447,6 +491,14 @@ int32_t random_dem_test(ptrdiff_t dims[2], uint32_t seed) {
   ptrdiff_t *prev = new ptrdiff_t[dims[0] * dims[1]];
   assert(prev != NULL);
 
+  // Outputs for gradient8
+  float *gradient = new float[dims[0] * dims[1]];
+  assert(flats != NULL);
+  // TODO: random value for cellsize?
+  float cellsize = 30.0;
+  // TODO: should different units get tested?
+  char unit = 't';
```

Comment:
Right now, the values for `cellsize` and `unit` are fixed in the test. Since the unit conversion is based on the already computed tangent value, I don't think it's really necessary to test every `unit`. We could run multiple test with different `cellsize` values for each run.

Reply:
I think the important thing is to run it with a `cellsize` that is not equal to 1, just to make sure that the gradient actually takes into account the `cellsize`.